openeo/extra/job_management/__init__.py

```diff
@@ -552,6 +582,35 @@ def _job_update_loop(
             self.on_job_cancel(job, row)
 
 
+    def _process_result_queue(self, stats: Optional[dict] = None):
+        """
+        Process results from the result queue, update job statuses, and persist changes.
+
+        :param job_db: The job database interface to persist changes.
+        :param not_started: DataFrame containing jobs that are not yet started.
+        :param stats: Dictionary to track statistics.
+        """
+        while not self._result_queue.empty():
+            _log.info('Result queue waiting for tasks')
```

Comment:
This log message might be confusing. At this point there is a result available in the queue, and nothing to wait on. Also we're not waiting here for tasks, but the outcomes from starting jobs